File: app/src/main/java/com/jeeps/gamecollector/comparators/GameByPhysicalComparator.java
package com.jeeps.gamecollector.comparators;

import com.jeeps.gamecollector.model.Game;

import java.util.Comparator;

/**
 * Created by jeeps on 1/9/2018.
 */

public class GameByPhysicalComparator implements Comparator<Game> {

    private boolean desc;

    public GameByPhysicalComparator(boolean desc) {
        this.desc = desc;
    }

    public GameByPhysicalComparator() {
        this.desc = false;
    }

    @Override
    public int compare(Game game1, Game game2) {
        if (!desc)
            return Boolean.compare(game1.isPhysical(), game2.isPhysical());
        return Boolean.compare(game1.isPhysical(), game2.isPhysical()) * -1;
    }
}

File: app/src/main/java/com/jeeps/gamecollector/remaster/utils/comparators/GameByPhysicalComparator.kt
package com.jeeps.gamecollector.remaster.utils.comparators

import com.jeeps.gamecollector.remaster.data.model.data.games.Game

/**
 * Created by jeeps on 1/9/2018.
 */
class GameByPhysicalComparator : Comparator<Game> {
    private var desc: Boolean

    constructor(desc: Boolean) {
        this.desc = desc
    }

    constructor() {
        desc = false
    }

    override fun compare(game1: Game, game2: Game): Int {
        return if (!desc)
            game1.isPhysical.compareTo(game2.isPhysical)
        else
            game1.isPhysical.compareTo(game2.isPhysical) * -1
    }
}